pkg/hostagent/ipam.go

```diff
@@ -192,6 +192,7 @@ func (agent *HostAgent) deallocateMdIps(md *metadata.ContainerMetadata) {
 				continue
 			}
 
+			agent.log.Infof("deallocating IP: %v", ip.Address.IP)
```

Comment:
It might be helpful to make logging of the deallocateMdIps function consistent with this function (i.e., the before and after logs)

Reply:
Also, why is it that in this case iface.IPs was not being set to nil as is being in L178?

Reply:
Afaik, we deallocate with  deallocateMdIps when the containers metadata exists, the calling function here takes care of clearing stale metadata, (ContainerIfaceMd, epMetadata)

On the other hand when the container metadata doesn't exist, we are setting it to nil manually, i.e. the function deallocateIpsLocked is called when for some reason an IP may have been assigned but the container fails to come up for other reasons.